### PR TITLE
Switch to Github Actions

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
_Please replace this text with a description of what this PR accomplishes._

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Unit Tests
+        uses: cedrickring/golang-action@1.5.2
+        env:
+          GO111MODULE: "on"
+        with:
+          args: make install && make test_unit_codecov
+      - name: Push CodeCov
```
Comment (kumare3) [inline .github/workflows/master.yml:77]: we will have to change make test_unit_codecov right?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        with:
+          username: "${{ github.actor }}"
+          password: "${{ secrets.GITHUB_TOKEN }}"
+          image_name: ${{ secrets.flytegithub_repo }}/flyteadmin
+          image_tag: latest,${{ github.sha }},${{ needs.bump-version.outputs.version }}
+          push_git_tag: true
+          registry: docker.pkg.github.com
+          build_extra_args: "--compress=true"
+      - uses: engineerd/setup-kind@v0.4.0
+      - name: End2End
```
Comment (kumare3) [inline .github/workflows/master.yml:39]: I think we should always remember that end-to-end tests should be smoke tests only - https://testing.googleblog.com/2015/04/just-say-no-to-more-end-to-end-tests.html#:~:text=While%20relying%20primarily%20on%20end,idea%20makes%20sense%20in%20theory.&text=Managers%20and%20decision%2Dmakers%20like,test%20would%20impact%20the%20user.

File: .github/workflows/master.yml (hunk 1 above)
Comment (EngHabu) [reply .github/workflows/master.yml:77]: yes. that'd done as part of this.

File: .github/workflows/master.yml (hunk 2 above)
Comment (EngHabu) [reply .github/workflows/master.yml:39]: Agreed..

**Pull request merged**